lib/trino-hive-formats/src/test/java/io/trino/hive/formats/avro/TestAvroPageDataReaderWithTypeManagement.java

```diff
@@ -0,0 +1,217 @@
+        large.addToSchema(DECIMAL_LARGE_BYTES_SCHEMA);
+        DECIMAL_LARGE_FIXED_SCHEMA = Schema.createFixed("largeDecimal", "myFixed", "namespace", (int) ((MAX_SHORT_PRECISION + 2) * Math.log(10) / Math.log(2) / 8) + 1);
+        large.addToSchema(DECIMAL_LARGE_FIXED_SCHEMA);
+    }
+
+    @Test
+    public void testTypesSimple()
+            throws IOException, AvroTypeException
+    {
+        Schema schema = SchemaBuilder.builder()
+                .record("allSupported")
+                .fields()
+                .name("timestampMillis")
+                .type(TIMESTAMP_MILLIS_SCHEMA).noDefault()
+                .name("timestampMicros")
+                .type(TIMESTAMP_MICROS_SCHEMA).noDefault()
+                .name("smallBytesDecimal")
+                .type(DECIMAL_SMALL_BYTES_SCHEMA).noDefault()
+                .name("smallFixedDecimal")
+                .type(DECIMAL_SMALL_FIXED_SCHEMA).noDefault()
+                .name("largeBytesDecimal")
+                .type(DECIMAL_LARGE_BYTES_SCHEMA).noDefault()
+                .name("largeFixedDecimal")
+                .type(DECIMAL_LARGE_FIXED_SCHEMA).noDefault()
+                .name("date")
+                .type(DATE_SCHEMA).noDefault()
+                .name("timeMillis")
+                .type(TIME_MILLIS_SCHEMA).noDefault()
+                .name("timeMicros")
+                .type(TIME_MICROS_SCHEMA).noDefault()
+                .name("id")
+                .type(UUID_SCHEMA).noDefault()
+                .endRecord();
+
+        GenericData.Fixed genericSmallFixedDecimal = new GenericData.Fixed(DECIMAL_SMALL_FIXED_SCHEMA);
+        genericSmallFixedDecimal.bytes(padBigEndianCorrectly(78068160000000L, SMALL_FIXED_SIZE));
+        GenericData.Fixed genericLargeFixedDecimal = new GenericData.Fixed(DECIMAL_LARGE_FIXED_SCHEMA);
+        genericLargeFixedDecimal.bytes(padBigEndianCorrectly(78068160000000L, LARGE_FIXED_SIZE));
+        UUID id = UUID.randomUUID();
+
+        GenericData.Record myRecord = new GenericData.Record(schema);
+        myRecord.put("timestampMillis", testTime.getTime());
+        myRecord.put("timestampMicros", testTime.getTime() * 1000);
+        myRecord.put("smallBytesDecimal", ByteBuffer.wrap(Longs.toByteArray(78068160000000L)));
+        myRecord.put("smallFixedDecimal", genericSmallFixedDecimal);
+        myRecord.put("largeBytesDecimal", ByteBuffer.wrap(Int128.fromBigEndian(Longs.toByteArray(78068160000000L)).toBigEndianBytes()));
+        myRecord.put("largeFixedDecimal", genericLargeFixedDecimal);
+        myRecord.put("date", 9035);
+        myRecord.put("timeMillis", 39_600_000);
+        myRecord.put("timeMicros", 39_600_000_000L);
+        myRecord.put("id", id.toString());
+
+        TrinoInputFile input = createWrittenFileWithData(schema, ImmutableList.of(myRecord));
+        try (AvroFileReader pageIterator = new AvroFileReader(input, schema, new NativeLogicalTypesAvroTypeManager())) {
```

Comment:
rename `pageIterator` everywhere to something like `avroFileReader` or `fileReader`